src/expr/src/relation/func.rs

```diff
@@ -534,6 +534,55 @@ fn lag_lead_inner<'a>(
     args: Vec<(Datum<'a>, Datum<'a>, Datum<'a>)>,
     lag_lead_type: &LagLeadType,
     ignore_nulls: &bool,
+) -> Vec<Datum<'a>> {
+    if *ignore_nulls {
+        lag_lead_inner_ignore_nulls(args, lag_lead_type)
+    } else {
+        lag_lead_inner_respect_nulls(args, lag_lead_type)
+    }
+}
+
+fn lag_lead_inner_respect_nulls<'a>(
+    args: Vec<(Datum<'a>, Datum<'a>, Datum<'a>)>,
+    lag_lead_type: &LagLeadType,
+) -> Vec<Datum<'a>> {
+    let mut result: Vec<Datum> = Vec::with_capacity(args.len());
+    for (idx, (_, offset, default_value)) in args.iter().enumerate() {
+        // Null offsets are acceptable, and always return null
+        if offset.is_null() {
+            result.push(*offset);
```

Comment:
This confused me a bit, I would expect spelling out `Datum::Null` here

Reply:
Changed.